GH #228 Skip tests with CHCKCLNT equal MQCHK_REQUIRED

SeyfSV opened the pull request:
https://github.com/dsuch/pymqi/issues/288#issuecomment-1114125532
> The connauth object USE.PW is configured to have CHCKCLNT set to OPTIONAL. This means if a username / password is provided in an MQCSP object, then it will be checked. But an application is not required to provide one. Therefore, this test should not be getting MQRC_NOT_AUTHORIZED in the default configuration provided by tox.ini.
Restricting the test to only run when USE.PW is configured to CHCKCLNT(REQUIRED) should keep the intended spirit of the test, while ensuring that it doesn't fail in other circumstances.

SeyfSV commented on code/tests/test_queue_manager.py:
```diff
@@ -5,87 +5,153 @@
             qmgr.disconnect()
 
-    def test_connect_tcp_client_conection_list(self):
+    # This test should be run last
```
@JoshMcMQ , maybe you have some idea about such behavior

Skyler-Altol replied:
It's currently midnight here so I'm a bit tired, but I may also be able to shed some light on this behaviour. I will revisit this tomorrow morning :)

Skyler-Altol commented:
@SeyfSV small error, the relevant QM parameter is not CHCKLOCL, but rather CHCKCLNT. The following test is connecting through a client connection rather than a bindings connection, so the value of CHCKLOCL has no effect on it:

https://github.com/SeyfSV/pymqi/blob/gh288-skip-chcklocal/code/tests/test_queue_manager.py#L134

JoshMcMQ commented:
Even with OPTIONAL if a MQCSP with username and password is provided it must be correct.  OPTIONAL allows you to send no username and password.  REQUIRED means you must send username and password.  REQDADM means of it is a admin user it behaves as REQUIRED, otherwise it behaves as OPTIONAL.  NONE means if a username and password is sent it is ignored.

Skyler-Altol commented:
> Even with OPTIONAL if a MQCSP with username and password is provided it must be correct. OPTIONAL allows you to send no username and password. REQUIRED means you must send username and password. REQDADM means if it is a admin user it behaves as REQUIRED, otherwise it behaves as OPTIONAL. NONE means if a username and password is sent it is ignored.

Hey Josh, I'm well aware. But CHCKLOCL applies to bindings (or local) connections, and CHCKCLNT applies to client connections over TCP. Skipping TCP based tests over the value set for CHCKLOCL wouldn't make sense.

Skyler-Altol commented:
> Even with OPTIONAL if a MQCSP with username and password is provided it must be correct. OPTIONAL allows you to send no username and password. REQUIRED means you must send username and password. REQDADM means if it is a admin user it behaves as REQUIRED, otherwise it behaves as OPTIONAL. NONE means if a username and password is sent it is ignored.

This is all true yes, but the point I was making is that when set to optional, it is completely valid to not present credentials. Hence a test that is expecting MQRC_NOT_AUTHORISED if credentials are invalid/not provided, is going to get an invalid result with that configuration :)

JoshMcMQ commented:
If invalid credentials are being presented (MQCSP  with MQCSP_AUTH_USER_ID_AND_PWD) then the only time you "may" not get MQRC_NOT_AUTHORISED is with OPTIONAL, but that depends on other factors, you could still get a MQRC_NOT_AUTHORISED  if the SVRCONN has a invalid user or a user without permissions is set in the MCAUSER or due to CHLAUTH rules

SeyfSV commented:
First:
@Skyler-Altol , it is not small error it is big fault :) Thank you. I don't know in what moment I switched from client to binding. - Fixed

Second:
I think you mentioned this test.
```
    def test_connect_tcp_client_without_cred(self):
        if self.CHCKLOCL == pymqi.CMQCFC.MQCHK_REQUIRED:
            self.skipTest('Test not viable for user/password-requiring queue manager')
            return

        qmgr = pymqi.QueueManager(None)
        with self.assertRaises(pymqi.MQMIError) as ex_ctx:
            qmgr.connect_tcp_client(
                self.queue_manager, pymqi.cd(), self.channel, self.conn_info)
            self.assertEqual(ex_ctx.exception.reason, pymqi.CMQC.MQRC_NOT_AUTHORIZED)
        if qmgr.is_connected:
            qmgr.disconnect()
```
What do you say if we will skip it in case of `self.CHCKLOCL != pymqi.CMQCFC.MQCHK_REQUIRED`?

Skyler-Altol commented:
> First: @Skyler-Altol , it is not small error it is big fault :) Thank you. I don't know in what moment I switched from client to binding. - Fixed
> 
> Second: I think you mentioned this test.
> 
> ```
>     def test_connect_tcp_client_without_cred(self):
>         if self.CHCKLOCL == pymqi.CMQCFC.MQCHK_REQUIRED:
>             self.skipTest('Test not viable for user/password-requiring queue manager')
>             return
> 
>         qmgr = pymqi.QueueManager(None)
>         with self.assertRaises(pymqi.MQMIError) as ex_ctx:
>             qmgr.connect_tcp_client(
>                 self.queue_manager, pymqi.cd(), self.channel, self.conn_info)
>             self.assertEqual(ex_ctx.exception.reason, pymqi.CMQC.MQRC_NOT_AUTHORIZED)
>         if qmgr.is_connected:
>             qmgr.disconnect()
> ```
> 
> What do you say if we will skip it in case of `self.CHCKLOCL != pymqi.CMQCFC.MQCHK_REQUIRED`?

If that was `self.CHCKCLNT != pymqi.CMQCFC.MQCHK_REQUIRED` then yeah that'd be quite valid @SeyfSV :)
